src/models/matches.rs

```diff
@@ -80,6 +118,113 @@ impl Match {
+          // Add the comma to the associated matches
+          self.associated_comma = Some(Range::from(sibling.range()));
+          current_node = sibling;
+          found_comma = true;
+          continue; // Continue the inner loop (i.e. evaluate next sibling)
+        } else if self._is_comment_safe_to_delete(&sibling, node, piranha_arguments, trailing) {
+          // Add the comment to the associated matches
+          self.associated_comments.push(Range::from(sibling.range()));
+          current_node = sibling;
+          found_comment = true;
+          continue; // Continue the inner loop (i.e. evaluate next sibling)
+        }
+        break; // Break the inner loop
+      }
+
+      let parent = current_node.parent();
+      // If buf is <0 or we have found a comment and a comma, we break
+      if buf < 0 || (found_comma && found_comment) || parent.is_none() {
+        break; // Break the outer loop
+      }
+      current_node = parent.unwrap();
+      buf -= 1;
+      continue; // Continue the outer loop (i.e. lookup parent's siblings for comma/comment)
+    }
+  }
+
+  /// Checks if the given node kind is a comment in the language (determined from piranha arguments)
+  fn is_comment(&self, kind: String, piranha_arguments: &PiranhaArguments) -> bool {
+    *piranha_arguments.cleanup_comments()
+      && piranha_arguments.language().comment_nodes().contains(&kind)
+  }
+
+  /// Checks if the given comment is safe to delete.
+  fn _is_comment_safe_to_delete(
+    &mut self, comment: &Node, deleted_node: &Node, piranha_arguments: &PiranhaArguments,
+    trailing: bool,
+  ) -> bool {
+    // Check if the comment is a comment in the language
+    if !self.is_comment(comment.kind().to_string(), piranha_arguments) {
+      return false;
+    }
+    // If trailing, check if the comment is on the same line as the deleted node
+    // i.e. where the deleted node ends or starts
+    let is_on_same_line = comment.range().start_point.row == deleted_node.range().end_point.row
+      || comment.range().start_point.row == deleted_node.range().start_point.row;
+
+    // If trailing, return if the comment is on the same line as the deleted node
+    if trailing {
+      return is_on_same_line;
+    }
+    // If not trailing, return if the comment is on the same line as the deleted node
+    if is_on_same_line {
+      return true;
+    }
+    // Check if the previous node does not overlap with the comment
+    if let Some(previous_node) = comment.prev_sibling() {
+      if self.overlaps(comment, &previous_node) {
+        return false;
+      }
+    }
+    // Check if there exists no node between the comment and the deleted node
+    if let Some(next_node) = comment.next_sibling() {
+      if next_node.start_byte() < deleted_node.start_byte() {
+        let (start_range, _) = self.get_first_and_last_associated_ranges();
+        if next_node.start_byte() < start_range.start_byte {
+          return false;
```

Comment:
When do we hit this case? I thought we were going backwards from the deleted node and a) adding every comment node to the list used to calculate `get_first_and_last_associated_ranges()`, b) stopping on the first non-comment (non-comma) node we see. So we should never be calling this method with a non-comment node between the comment and the start of the current expanded range, no? What am I missing?

Reply:
Ah, basically, this exists because we jump to the parent without checking that the node being inspected is the first (non-comment) node inside said parent, correct?

Reply:
yes correct. We do not do that check instead we have the check we are discussing above.